rkt/image/namefetcher.go

```diff
@@ -183,31 +183,39 @@ func (f *nameFetcher) fetchVerifiedURL(app *discovery.App, u *url.URL, a *asc) (
 	o := f.httpOps()
 	ascFile, retry, err := o.DownloadSignature(a)
 	if err != nil {
-		return nil, nil, err
+		return NopReadSeekCloser(nil), nil, err
 	}
-	defer func() { maybeClose(ascFile) }()
+	defer func() {
+		if ascFile != nil {
+			ascFile.Close()
+		}
+	}()
```

Comment:
As explained above `DownloadSignature` now maintains the invariant of returning a closeable in case of no error, hence just `defer ascFile.Close()`.